tools/parse/src/main/scala/org/allenai/nlpstack/parse/poly/fsm/FSMTrainingVectorSource.scala

```diff
@@ -34,7 +36,7 @@ abstract class FSMTrainingVectorSource(
     if (!vectorIter.hasNext) {
       resultSoFar
     } else {
-      taskHelper(resultSoFar + vectorIter.next.task, vectorIter)
+      taskHelper(resultSoFar + vectorIter.next().task, vectorIter)
     }
```

Comment:
It's better when you have a large set of training vectors that are being read from disk. With getVectorIterator.map(_.task).toSet, you need to hold K tasks in memory at some point, where K is the number of training vectors. With this method, you only have to store the number of unique tasks, which is typically much much smaller.